### PR TITLE
Feat/feature support set dimension index api

### LINKED ISSUE #170: [Feature] support setDimensionIndex api to enable user trigger dimension effect
### What problem does this feature solve?

support user trigger dimension by api

### What does the proposed API look like?

vchart.setDimensionIndex(value: StringOrNumber)

### PR DESCRIPTION
**Pull request by xuefei1313**
[[中文版模板 / Chinese template](https://github.com/VisActor/VChart/blob/main/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Refactoring
- [ ] Update dependency
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Release
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Workflow
- [ ] Other (about what?)

### 🐞 Bugserver case id

close #170

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### 🚀 Summary

copilot:summary

### 🔍 Walkthrough

copilot:walkthrough

### CODE REVIEW

File: packages/vchart/src/chart/base-chart.ts [hunk 1]
```diff
@@ -1241,4 +1249,58 @@ export class BaseChart extends CompilableBase implements IChart {
       }
     });
   }
+
+  /**
+   * setDimensionIndex could trigger make state, tooltip, crosshair
```
Comment (kkxxkk2019) [inline packages/vchart/src/chart/base-chart.ts:1254]: mark

File: packages/vchart/src/core/vchart.ts [hunk 2]
```diff
@@ -1047,4 +1047,13 @@ export class VChart implements IVChart {
...
     return this._chart.getAllComponents();
   }
+
+  /**
+   * setDimensionIndex could trigger make state, tooltip, crosshair
+   * @param value dimension value
+   * @param opt option for set trigger
+   */
+  setDimensionIndex(value: StringOrNumber, opt: DimensionIndexOption = {}) {
+    return this._chart.setDimensionIndex(value, opt);
```
Comment (kkxxkk2019) [inline packages/vchart/src/core/vchart.ts:1057]: 需要判断下 this._chart 是否存在

File: packages/vchart/src/event/event.ts [hunk 3]
```diff
@@ -21,6 +21,9 @@ export class Event implements IEvent {
 
   private _composedEventMap: Map<EventCallback<EventParams>, { eventType: EventType; event: IComposedEvent }> =
     new Map();
+  composedEventMap() {
```
Comment (kkxxkk2019) [inline packages/vchart/src/event/event.ts:24]: getComposedEventMap

File: packages/vchart/src/event/events/dimension/base.ts [hunk 4]
```diff
@@ -44,4 +46,33 @@ export class DimensionEvent implements IComposedEvent {
 
     return result;
   }
+
+  dispatch(v: unknown, opt: { filter?: (axis: IAxis) => boolean }) {
+    // get all enable axis
```
Comment (kkxxkk2019) [inline packages/vchart/src/event/events/dimension/base.ts:51]: 需要处理下 this._chart 空吗

**Review by zamhown — APPROVED** [review]

**Review by kkxxkk2019 — APPROVED** [review]

**Pull request merged**